torch/onnx/symbolic_opset9.py

```diff
@@ -2745,3 +2745,13 @@ def as_strided(g, self, sizes, strides, offset=None):
         if offset:
             ind = g.op("Add", ind, g.op("Constant", torch.tensor([offset])))
         return g.op("Gather", self_1d, ind)
+
+
+def __derive_index(g, index, start, step):
+    return g.op("Add", start, g.op("Mul", index, step))
+
+
+def __range_length(g, lo, hi, step):
+    sub = g.op("Sub", hi, lo)
+    div = g.op("Ceil", true_divide(g, sub, step))
```

Comment:
if `sub` can be divided by `step`, would it be a `+1` there? for example, `sub=8`, `step=2`, then should be 5 elements not 4, right?

Reply:
this depends on if `hi` is inclusive right? if inclusive `+1` is needed. It will be helpful if link to doc/spec of these two ops can be added as comment here.

Reply:
Correct, as @jiafatom mentioned, this range in inclusive. So a +1 is needed if sub is divisible by step. But the Ceil node (on floats) is taking care of this situation. I added more tests and comments.

Reply:
A little confused here - `range` seems exclusive, so my example above should be 4 elements, and we don't need `+1` - then `Ceil` takes care of this and this is indeed the implementation here, should be correct.

Reply:
Sorry for the confusion, this is correct.